docs/config/lua/wezterm.table/length.md

```diff
@@ -0,0 +1,26 @@
+# `wezterm.table.length(table)`
+
+{{since('nightly')}}
+
+This function returns the length of a Lua table (or array) passed to it.
+
+Note: The Lua function `#` also returns the length of an array, but
+`#` only works for array-style tables with contiguous integer keys starting with index `1`, and not sparse arrays (with gaps in their integer keys), or object or other style of tables with non-integer keys.
```

Comment:
Since Lua unsets elements of a table by setting them equal to `nil`, I don't think we can get around this "unexpected" behavior (from my point of view I think this should be the expected behavior in Lua). I think I will just need to write better docs here and clarify what it counts (plus probably rename it to `count` as wez suggested).